x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -25,7 +25,11 @@ export function AnomalyDetectionSetupLink() {
 
   // Show alert if there are no jobs OR if no job matches the current environment
   const showAlert =
-    isFetchSuccess && !data.jobs.some((job) => environment === job.environment);
+    isFetchSuccess &&
+    !environmentIncludes(
+      data.jobs.map((job) => job.environment),
+      environment
+    );
```

Comment:
What was the problem with the previous check?

Reply:
the logic was wrong, I wrote unit tests, and it failed the one where environment was undefined and there were existing jobs. so it was showing the alert, when it should not have shown it.

Reply:
WDYT?

```ts

const showAlert =
  isFetchSuccess &&

  // "All environments" selected and at least one ML job exists
  ((!environment && data.jobs > 0) ||

  // a job exists for the selected environment
  !data.jobs.some((job) => environment === job.environment));
```